src/server/spell.js

```diff
@@ -31,16 +31,23 @@ class Spell {
       y: Math.floor(Constants.SPELL_GRID_SIZE / 2)
     };
 
+    let movement = null;
```

Comment:
Could we call it `lastMovement` or `lastDirection` to make the name of the variable more self-explanatory?

Reply:
Sure. `if (movement)` in the loop is not correct. I have renamed `movement` to `lastMovement` and reintroduced the `movement` back inside the loop.